### PR TITLE
Added 'delete' command in 'tink hardware' cli

### PR DESCRIPTION
**Pull request by parauliya**
Now `tink hardware` cli will have `delete` command which works based on `hardware-id` as follows:

   `tink hardware delete <hardware-id>`

This will enable the functionality of deleting a hardware.

### CODE REVIEW

File: cmd/tink-cli/cmd/hardware/delete.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	"github.com/spf13/cobra"
+	"github.com/tinkerbell/tink/client"
+	"github.com/tinkerbell/tink/protos/hardware"
+)
+
+// idCmd represents the id command
+var deleteCmd = &cobra.Command{
+	Use:     "delete",
+	Short:   "Delete hardware by id",
```
[the remark this answers is not in the document]
Comment (parauliya) [reply cmd/tink-cli/cmd/hardware/delete.go:17]: In  `cmd/hardware/`, every description starts with Capital case. I will change them all.

File: protos/hardware/hardware.proto [hunk 2]
```diff
@@ -30,3 +31,7 @@ message GetRequest {
 message Hardware {
 	string JSON = 1;
 }
+
+message DeleteRequest {
+	string ID = 1;
+}
```
[the remark this answers is not in the document]
Comment (parauliya) [reply protos/hardware/hardware.proto:37]: Done

File: cmd/tink-cli/cmd/hardware/delete.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	Args: func(_ *cobra.Command, args []string) error {
+		return verifyUUIDs(args)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		for _, id := range args {
+			_, err := client.HardwareClient.Delete(context.Background(), &hardware.DeleteRequest{ID: id})
+			if err != nil {
+				log.Fatal(err)
+			}
+			log.Println("Hardware data with id", id, "deleted successfully")
```
[the remark this answers is not in the document]
Comment (parauliya) [reply cmd/tink-cli/cmd/hardware/delete.go:28]: I followed the same pattern that you did in the push.go 
`log.Println("Hardware data pushed successfully")`. Should I fix that as well?

File: cmd/tink-cli/cmd/hardware/delete.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"context"
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/tinkerbell/tink/client"
+	"github.com/tinkerbell/tink/protos/hardware"
+)
+
+// idCmd represents the id command
```
[the remark this answers is not in the document]
Comment (parauliya) [reply cmd/tink-cli/cmd/hardware/delete.go:14]: Done

File: cmd/tink-cli/cmd/hardware/delete.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (gauravgahlot) [reply cmd/tink-cli/cmd/hardware/delete.go:28]: Ahh, got it. It's printed as info after the command completes. Let it be as is.

**Pull request merged**